### PR TITLE
Fix node and webpack-env compatibility

### LINKED ISSUE #9241: @storybook/react types dependency @types/webpack-env conflicting with @types/node
**Describe the bug**
`@storybook/react` types dependency `@types/webpack-env` conflicting with `@types/node`

```bash
node_modules/@types/webpack-env/index.d.ts:282:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'require' must be of type 'Require', but here has type 'NodeRequire'.
```
**To Reproduce**
1. Setup a TS env
2. Install @storybook/react
3. Install @types/node
4. Compile project

**Screenshots**
Screenshot 2019-12-24 at 4 27 34 PM

**System:**
```
System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-3635QM CPU @ 2.40GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Browsers:
    Chrome: 79.0.3945.88
    Safari: 13.0.3
```

**Additional context**
Manually editing @types/webpack-env solves this problem.
I tried adding "types" to tsconfig.json to limit type definition scope.
I am using storybook in monorepo setup using yarn.

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes https://github.com/storybookjs/storybook/issues/9241 (and build errors of our own) by making @types/webpack-env compatible with both v12 and v13 of @types/node. Unfortunately, I had to change which global interface `require` and `module` point at in node, which can change how packages that merge with Node’s globals can impact those types. @SimonSchick, does this look like a problem to you? Not sure what the motivation was for leaving `NodeRequire` around but changing `require` not to point to it.

### CODE REVIEW

**Comment by SimonSchick** [comment]
The motivation for the v13 changes was to cleanup globals that were not in the `NodeJS` namespace.
I left these aliases behind to provide backwards compatibility for other packages that relied on it.
What is the issue that requires the changes in the node globals?
Does decl merging not work as expected in the namespace?

**Comment by andrewbranch** [comment]
> I left these aliases behind to provide backwards compatibility for other packages that relied on it.

This backward compatibility doesn’t seem to do much if the type of `require` is now `NodeJS.Require`. So declaration merging into `NodeRequire` technically is still happening, but it has no observable effect because nothing references `NodeRequire`. And if the package re-declares the global `var require: NodeRequire` (like webpack-env does), it fails to compile because that’s not the same type as @types/node says `require` is.

**Comment by andrewbranch** [comment]
webpack-env can be updated to work with @types/node@13, but it was impossible to make it compatible with both v12 and v13 without this change to node. Since @types/webpack-env and @types/node are both extremely highly used, it would be nice for them to be cross-compatible at more than one version.

File: types/webpack-env/index.d.ts [hunk 1]
```diff
@@ -55,19 +59,17 @@ declare namespace __WebpackModuleApi {
...
         cache: {
-            [id: string]: any;
+            [id: string]: NodeModule;
         }
     }
 
     interface Module {
         exports: any;
-        require(id: string): any;
-        require<T>(id: string): T;
```
Comment (andrewbranch) [inline types/webpack-env/index.d.ts]: I couldn’t find from the webpack docs that this is real

File: types/webpack-env/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for webpack (module API) 1.14
+// Type definitions for webpack (module API) 1.15
```
Comment (andrewbranch) [inline types/webpack-env/index.d.ts]: Does this version track something external?

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**